crypto/x509/x509type.c

```diff
@@ -41,6 +41,9 @@ int X509_certificate_type(const X509 *x, const EVP_PKEY *pkey)
     case EVP_PKEY_EC:
         ret = EVP_PK_EC | EVP_PKT_SIGN | EVP_PKT_EXCH;
         break;
+    case EVP_PKEY_ED25519:
+        ret = EVP_PKT_SIGN;
```

Comment:
X509_certificate_type() is an ancient function. The only reason Ed25519 has an entry there is because some code in tls_process_cert_verify() needs it to function. The only other recent addition (GOST) only has the capabilities bits set and no algorithm bits. Also I just checked EVP_PK_EC is not referenced anywhere in OpenSSL.

TBH while looking through this it was clear that adding a new certificate type was far more complex than it should be. There are functions all over the place that need changing and obscre additions to if/switch statements which it is too easy to get wrong. The only way to find them all is to test it and see what breaks and then fix it hoping you catch everything. 

I think the whole thing can be handled with a single table of certificate capabilities. I'd planned a follow up PR to do exactly that. I could instead do that first and convert this PR to use the new scheme.